timemachine/fe/atom_mapping.py

```diff
@@ -330,12 +343,12 @@ def _get_cores_impl(
     )
 
     if connected_core:
```

Comment:
Two notes:


1) I think we should just expand the behavior here explicitly after today's discussion (since we may need a case where complete_rings=True but connected_core=False)

So I think roughly

```

if connected_core and complete_rings:
        # order matters here
        all_cores, all_marcs = remove_incomplete_rings(mol_a, mol_b, all_marcs, all_cores)
        all_cores, all_marcs = remove_disconnected_components(mol_a, mol_b, all_cores, all_marcs)
elif connected_core and not complete_rings:
        all_cores, all_marcs = remove_disconnected_components(mol_a, mol_b, all_cores, all_marcs)
elif completed_rings and not connected_core:
        all_cores, all_marcs = remove_incomplete_rings(mol_a, mol_b, all_marcs, all_cores)
else:
        pass
```

2) Let's make the arguments of `remove_disconnected_components` and `remove_incomplete_rings` consistent, i.e. either have them both be `mol_a, mol_b, all_cores, all_marcs` or  both `mol_a, mol_b, all_marcs, all_cores`

Reply:
updated